python/cudf/cudf/fuzz_tests/fuzz_test_parquet.py

```diff
@@ -38,5 +67,44 @@ def parquet_writer_test(gdf):
     assert_eq(actual, expected)
 
 
+@pythonfuzz(
+    data_handle=ParquetWriter,
+    params={
+        "row_group_size": np.random.random_integers(0, 10000, 100),
+        "compression": ["snappy", None],
+        "index": [True, False, None],
+    },
+)
+def parquet_writer_test_rowgroup_index_compression(
+    gdf, index, compression, row_group_size
+):
+    print("PARAMS", index, compression, row_group_size)
+    pd_file_name = "cpu_pdf.parquet"
+    gd_file_name = "gpu_pdf.parquet"
+
+    pdf = gdf.to_pandas()
+
+    pdf.to_parquet(
+        pd_file_name,
+        index=index,
+        compression=compression,
+        row_group_size=row_group_size,
+    )
+    gdf.to_parquet(
+        gd_file_name,
+        index=index,
+        compression=compression,
+        row_group_size=row_group_size,
+    )
+
+    actual = cudf.read_parquet(gd_file_name)
+    expected = pd.read_parquet(pd_file_name)
+    assert_eq(actual, expected)
+
+    actual = cudf.read_parquet(pd_file_name)
+    expected = pd.read_parquet(gd_file_name)
+    assert_eq(actual, expected)
```

Comment:
Similarly here, using `gdf.to_pandas()` removes nullable integer / nullable bool / etc. columns and converts to float. Can we use PyArrow directly?

Reply:
Yes, this is now fixed by using an internal utility function that preserves pandas nullable dtypes.